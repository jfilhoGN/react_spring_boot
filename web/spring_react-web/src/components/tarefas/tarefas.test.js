import React from 'react';
import {shallow} from 'enzyme';
import sinon from 'sinon';
import toJson from 'enzyme-to-json';
import Tarefas from './tarefas';

describe('Tarefas', () => {
    test('render component', () => {
        const history = {
            goBack: jest.fn(),
        }
        global.sessionStorage.getItem.mockReturnValue('{"id": "28A", "token":"eyJ0eX"}')
        const wrapper = shallow(<Tarefas history={history}/>);
        expect(wrapper).toMatchSnapshot();
    });

    test('values state', () => {
        const history = {
            goBack: jest.fn(),
        }
        const wrapper = shallow(<Tarefas history={history}/>);
        expect(wrapper.state).toMatchSnapshot();
    });

    test('getTarefas()', () => {
        const history = {
            goBack: jest.fn(),
        }
        const wrapper = shallow(<Tarefas history={history}/>);
        //terminar
    });

    test('deleteClick()', () => {
        //arrumar
    });

    test('editClick()', () => {
        //arrumar
    });

    test('searchFilterFunction', () => {
        //arrrumar
    });

    test('changeStatus()', () => {
        //arrumar
    });

    test('renderTarefas()', () => {
        //arrumar
    });

});