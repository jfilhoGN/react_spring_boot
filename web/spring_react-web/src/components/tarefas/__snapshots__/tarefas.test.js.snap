// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tarefas render component 1`] = `
<Card
  className="card-tarefas"
>
  <Row>
    <Button
      className="btn waves-effect waves-light btn-small grey"
      name="action"
      node="button"
      onClick={[Function]}
      title="adicionar tarefa"
      type="submit"
    >
      <i
        className="material-icons"
      >
        add
      </i>
    </Button>
    <div>
      <form>
        <input
          onChange={[Function]}
          placeholder="procurar tarefa..."
        />
      </form>
    </div>
    <Col
      className="grey darken-3 "
      m={12}
      s={3}
    >
      <Collection
        className="collection"
      />
    </Col>
    <h3
      className="h3-tarefas"
    >
      <center>
        Tarefas
      </center>
    </h3>
    <Col
      className="grey darken-3 "
      m={3}
      s={3}
    >
      <center>
        <h4
          className="h4-tarefas"
        >
          Fazer
        </h4>
      </center>
      <Collection
        className="collection"
      />
    </Col>
    <Col
      className="grey darken-3"
      m={3}
      s={3}
    >
      <center>
        <h4
          className="h4-tarefas"
        >
          Fazendo
        </h4>
      </center>
      <Collection
        className="collection"
      />
    </Col>
    <Col
      className="grey darken-3"
      m={3}
      s={3}
    >
      <center>
        <h4
          className="h4-tarefas"
        >
          Teste
        </h4>
      </center>
      <Collection
        className="collection"
      />
    </Col>
    <Col
      className="grey darken-3"
      m={3}
      s={3}
    >
      <center>
        <h4
          className="h4-tarefas"
        >
          Concluido
        </h4>
      </center>
      <Collection
        className="collection"
      />
    </Col>
  </Row>
</Card>
`;

exports[`Tarefas values state 1`] = `[Function]`;
